test/evergreen.yml

```diff
@@ -6479,13 +6491,24 @@ buildvariants:
     CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/clang.cmake
     CMAKE_BUILD_TYPE: -DCMAKE_BUILD_TYPE=ASan
     ENABLE_TCMALLOC: 0
+    # PYTHONMALLOC is key to avoiding a long suppression list when running python under ASan. It
+    # turns off python's own allocators and uses the default system allocators.
+    # Then ASan symbolizer is intentionally set to toolchain v4 because the v5 symbolizer leaks
+    # memory which triggers recursion.
     additional_env_vars: |
       export LSAN_OPTIONS="$COMMON_SAN_OPTIONS:print_suppressions=0:suppressions=$(git rev-parse --show-toplevel)/test/evergreen/asan_leaks.supp"
+      export ASAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v4/bin/llvm-symbolizer
+      export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$(dirname $(/opt/mongodbtoolchain/v5/bin/clang -print-file-name=libclang_rt.asan.so))"
+      export PYTHONMALLOC=malloc
     num_jobs: $(echo "`grep -c ^processor /proc/cpuinfo`" | bc)
+    # ASan preload turns on LD_PRELOAD prior to execution of a python unit test. The reason we don't
+    # export LD_PRELOAD as part of additional_env_vars is that it breaks _everything_.
+    asan_preload: true
     # Exclude wt12015_backup_corruption for ASan builds. It creates false positives when it deliberately kills processes that invoke corruption.
     ctest_extra_args: -E "wt12015_backup_corruption"
   tasks:
     - name: ".pull_request !.pull_request_compilers !.python !.tiered_unittest !csuite-wt12015-backup-corruption-test"
+    - name: ".unit_test"
```

Comment:
I'm intentionally using the ".unit_test" tag here as it will give us access to the "unit_test_bucketXX" tasks, which running under asan are reasonably slower so I figured having them split was convenient.